subcommand/server-acl-init/command.go

```diff
@@ -522,42 +684,68 @@ func (c *Command) configureConnectInject(logger hclog.Logger, consulClient *api.
 		BindName:    "${serviceaccount.name}",
 		Selector:    c.flagBindingRuleSelector,
 	}
-	c.untilSucceeds(fmt.Sprintf("creating acl binding rule for %s", authMethodTmpl.Name),
+	return c.untilSucceeds(fmt.Sprintf("creating acl binding rule for %s", authMethodTmpl.Name),
 		func() error {
 			_, _, err := consulClient.ACL().BindingRuleCreate(&abr, nil)
 			return err
 		}, logger)
 }
 
-func (c *Command) untilSucceeds(opName string, op func() error, logger hclog.Logger) {
+// untilSucceeds runs op until it returns a nil error.
+// If c.cmdTimeout is cancelled it will exit.
+func (c *Command) untilSucceeds(opName string, op func() error, logger hclog.Logger) error {
 	for {
 		err := op()
 		if err == nil {
 			logger.Info(fmt.Sprintf("Success: %s", opName))
 			break
 		}
 		logger.Error(fmt.Sprintf("Failure: %s", opName), "err", err)
-		logger.Info("Retrying in 1s")
-		time.Sleep(1 * time.Second)
+		logger.Info("Retrying in " + c.retryDuration.String())
+		// Wait on either the retry duration (in which case we continue) or the
+		// overall command timeout.
+		select {
+		case <-time.After(c.retryDuration):
+			continue
+		case <-c.cmdTimeout.Done():
+			return errors.New("reached command timeout")
+		}
 	}
+	return nil
 }
 
+// isNoLeaderErr returns true if err is due to trying to call the
+// bootstrap ACLs API when there is no leader elected.
 func isNoLeaderErr(err error) bool {
 	return err != nil && strings.Contains(err.Error(), "Unexpected response code: 500") &&
 		strings.Contains(err.Error(), "The ACL system is currently in legacy mode.")
 }
 
+// isPolicyExistsErr returns true if err is due to trying to call the
+// policy create API when the policy already exists.
 func isPolicyExistsErr(err error, policyName string) bool {
 	return err != nil &&
 		strings.Contains(err.Error(), "Unexpected response code: 500") &&
 		strings.Contains(err.Error(), fmt.Sprintf("Invalid Policy: A Policy with Name %q already exists", policyName))
 }
 
-const synopsis = "Initialize ACLs on Consul servers."
+// podAddr is a convenience struct for passing around pod names and
+// addresses for Consul servers.
+type podAddr struct {
+	// Name is the name of the pod.
+	Name string
+	// Addr is in the form "<ip>:<port>".
+	Addr string
+}
```

Comment:
Add `Scheme` attribute to allow for https scheme consul client to be assigned on the getConsulServers above.